cdx-cli/db-setup/jit-workload-account/6.update-tags.sh

```diff
@@ -0,0 +1,217 @@
+
+    if [ -n "$REPO_ARN" ] && [ "$REPO_ARN" != "None" ]; then
+        aws ecr tag-resource --resource-arn "$REPO_ARN" \
+            --tags "Key=Name,Value=${repo}" "Key=purpose,Value=database-iam-jit" "Key=created_by,Value=cloudanix" || \
+            log "Warning: Failed to tag ECR repository $repo"
+        log "Tagged ECR repository: $repo"
+    else
+        log "Warning: Repository $repo not found!"
+    fi
+done
+
+# Find and tag Internet Gateway
+log "Finding and tagging Internet Gateway..."
+IGW_ID=$(aws ec2 describe-internet-gateways \
+    --filters "Name=attachment.vpc-id,Values=${VPC_ID}" \
+    --query 'InternetGateways[0].InternetGatewayId' \
+    --output text)
+if [ "$IGW_ID" != "None" ]; then
+    add_tags "arn:aws:ec2:${AWS_REGION}:${ACCOUNT_ID}:internet-gateway/${IGW_ID}" "${PROJECT_NAME}-igw"
+fi
+
+# Find and tag NAT Gateways
+log "Finding and tagging NAT Gateways..."
+NAT_GATEWAYS=$(aws ec2 describe-nat-gateways \
+    --filter "Name=vpc-id,Values=${VPC_ID}" \
+    --query 'NatGateways[*].[NatGatewayId]' \
+    --output text)
+for nat_id in $NAT_GATEWAYS; do
+    if [ -n "$nat_id" ]; then
+        add_tags "arn:aws:ec2:${AWS_REGION}:${ACCOUNT_ID}:natgateway/${nat_id}" "${PROJECT_NAME}-nat"
+    fi
+done
+
+# Find and tag Route Tables
+log "Finding and tagging Route Tables..."
+ROUTE_TABLES=$(aws ec2 describe-route-tables \
+    --filters "Name=vpc-id,Values=${VPC_ID}" \
+    --query 'RouteTables[*].RouteTableId' \
+    --output text)
+for rt_id in $ROUTE_TABLES; do
+    if [ -n "$rt_id" ]; then
+        add_tags "arn:aws:ec2:${AWS_REGION}:${ACCOUNT_ID}:route-table/${rt_id}" "${PROJECT_NAME}-rt"
+    fi
+done
+
+# Find and tag EFS File Systems
+log "Finding and tagging EFS File Systems..."
+EFS_SYSTEMS=$(aws efs describe-file-systems \
+    --query 'FileSystems[*].[FileSystemId,Tags[?Key==`Name`].Value|[0]]' \
+    --output text)
+while IFS=$'\t' read -r efs_id efs_name; do
+    if [ -n "$efs_id" ] && [[ "$efs_name" == *"$PROJECT_NAME"* ]]; then
+        add_tags "arn:aws:elasticfilesystem:${AWS_REGION}:${ACCOUNT_ID}:file-system/${efs_id}" "${PROJECT_NAME}-efs"
+    fi
+done <<< "$EFS_SYSTEMS"
+
+# Tag CloudWatch Log Groups
+log "Tagging CloudWatch Log Groups..."
+LOG_GROUPS=$(aws logs describe-log-groups \
+    --log-group-name-prefix "/ecs/${PROJECT_NAME}" \
+    --query 'logGroups[*].logGroupName' \
+    --output text)
+for log_group in $LOG_GROUPS; do
+    aws logs tag-log-group \
+        --log-group-name "$log_group" \
+        --tags "purpose=database-iam-jit,created_by=cloudanix" || log "Warning: Failed to tag log group $log_group"
+done
+
+# Find and tag Secrets Manager secrets
+log "Finding and tagging Secrets Manager secrets..."
+SECRETS=$(aws secretsmanager list-secrets \
+    --query 'SecretList[?Name==`CDX_SECRETS`].ARN' \
+    --output text)
+for secret_arn in $SECRETS; do
+    if [ -n "$secret_arn" ] && [ "$secret_arn" != "None" ]; then
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code/findings/d803ac66c50fdc16/PR/1/artifacts/Divyansh/latest/secrets/show" target='_blank'>Click here for more details</a></b>